src/simulation/m_bubbles.fpp

```diff
@@ -79,16 +87,23 @@ contains
         !!  @param fR Current bubble radius
         !!  @param fV Current bubble velocity
         !!  @param fpb Internal bubble pressure
-    function f_cpbw(fR0, fR, fV, fpb)
+        !!  @param gam_l Physical Bubble parameter (pass in from global parameters)
+        !!  @param Ca_l Cavitation number (pass in from global parameters)
+        !!  @param Web_l Weber number (pass in from global parameters)
+        !!  @param Re_inv_l Inverse Reynolds number (pass in from global parameters)
+        !!  @param polytropic_l Polytropic  switch (pass in from global parameters)
+    pure elemental function f_cpbw(fR0, fR, fV, fpb, gam_l, Ca_l, Web_l, Re_inv_l, polytropic_l)
         !$acc routine seq
         real(wp), intent(in) :: fR0, fR, fV, fpb
+        real(wp), intent(in) :: gam_l, Ca_l, Web_l, Re_inv_l
+        logical, intent(in) :: polytropic_l
```

Comment:
Similar to above, since these are `intent(in)` why was the refactor necessary to make it `pure`?